app/controllers/concerns/sortable.rb

```diff
@@ -0,0 +1,24 @@
+module Sortable
+  extend ActiveSupport::Concern
+
+  def sort
+    sortable_class.transaction do
+      sort_params[:sortable_ids].each_with_index do |id, index|
+        sortable_class.where(id: id.to_i).update_all({ position: index + 1 })
```

Comment:
We need to scope tags by project, otherwise authors will be able to edit tags even for projects they dont have access to. I think the issue here is we're passing `sortable_class` to this concern but this concern does not have any concept of scoping. Instead, we should pass the collection. The TagsController/RulesController should already have the necessary context to build the `@tags` or `@rules` that we could use here.

Reply:
Updated to pull `sortable_records` from each controller like we're doing to get the class, so now we're limiting the list accordingly in each controller